### DISCUSSION TITLE
LSPosed version API

### DISCUSSION THREAD

**Discussion by Mikanoshi** [LSPosed/LSPosed#1626] [Q&A] [answered]
Is it possible to get current framework version for bug reporting needs?
If app is hooked it should have an Xposed API available that returns something like "LSPosed 1.7.1 (6358)".
Can't check for manager app version if it's parasitic, looks like _com.android.shell_ and _org.lsposed.manager_ are not even hookable to grab version from there. Can regular app get access to lspd? Module/verbose log should be available too, it is a nightmare to debug something without logs, and their automatic collection is a lot more preferable than forcing user to fill in everything manually, because they usually cannot be bothered.

**Comment by mywalkb** [comment, marked as answer]
I think will be possible only when #9 will be completed.
[Here](https://github.com/LSPosed/LSPosed/pull/1593/files#diff-9ee85419612299f4e66dd64bee62d018a955f951454a7960debc609f3c291744R45) there is the code for get the XposedService.
In [this](https://github.com/libxposed/XposedService/issues/1) issue there is some informations.

**Reply by Mikanoshi** [reply]
CLI requires root though? Can it work from system_server?

**Reply by mywalkb** [reply]
yes cli require root when the PR will be accepted, but that code was only an example. You can't get a ILSPManagerService with the same method of cli.
When #9 will be completed there will be a similar method I hope. Try to see [this](https://github.com/LSPosed/LSPosed/commit/10a2ae56a66607ff824f2cda3e71038f800a4beb) commit maybe is useful for get your information.
I don't know if for you is better an ILSPApplicationService, XposedService or ILSPManagerService.

> Can it work from system_server?

I don't know

**Reply by Mikanoshi** [reply]
Thanks. I have no idea how all this IPC works :) I guess I'll have to wait until it's released and available as API.

**Reply by mywalkb** [reply]
> I have no idea how all this IPC works

same for me. I developed some test code but without success. For ILSPManagerService I think will never available to xposed modules for security reasons.
